Extension/src/LanguageServer/extension.ts

```diff
@@ -248,6 +249,11 @@ export async function activate(): Promise<void> {
         clients.timeTelemetryCollector.setFirstFile(activeEditor.document.uri);
         activeDocument = activeEditor.document;
     }
+
+    if (util.extensionContext && new CppSettings().experimentalFeatures) {
```

Comment:
@benmcmorran 
this is not great because it will throw whenever the `#cpp` tool is invoked by the user on the Chat prompt:

![image](https://github.com/user-attachments/assets/f7699ffe-aa17-453b-8538-28576e90f00a)


Should a `when` clause added to `languageModelTools`?

Reply:
Good catch, I forgot these are statically registered now. A `when` clause makes sense to me.

Reply:
added a `when:` clause